Add the backup documentation for JDBC databases (#157)
scalar-labs/scalardb: docs/backup-restore.md

To minimize mistakes when doing backup and restore operations, it is recommended to use [Cassy](https://github.com/scalar-labs/cassy).
Cassy is also itegrated with `scalar-admin` so it can issue a pause request to the application of a Cassandra cluster.
Please see [the doc](https://github.com/scalar-labs/cassy/blob/master/docs/getting-started.md#take-cluster-wide-consistent-backups) for more details.

## JDBC databases

You can take a backup with your favorite way for the JDBC database that you are using.

> One requirement for backup in Scalar DB for JDBC databases is that the backup for all the tables (including the coordinator table) needs to be a consistent snapshot, which means all the backup data need to be taken at the same time.

Reviewer: ```suggestion
One requirement for backup in Scalar DB for JDBC databases is that backups for all the Scalar DB managed tables (including the coordinator table) need to be transactionally-consistent.
```

Reviewer: @brfrn169 Not fully sure if it is good enough, but I updated the sentence based on your feedback. PTAL.
